src/fpylll/algorithms/babai.py

```diff
@@ -46,15 +48,16 @@ def babai(B, t):
             A[i, j] = B[i, j]
 
     # make sure the input is LLL reduced before reading the norm of the last vector
-    LLL.reduction(A)
+    LLL.reduction(A, *args, **kwargs)
     # zero vector at the end
     A.swap_rows(0, B.nrows)
 
     for j in range(B.ncols):
         A[-1, j] = t[j]
-    A[-1, -1] = ceil(A[-2].norm())
+    # precise norm
+    A[-1, -1] = isqrt(sum(x**2 for x in A[-2])) + 1
```

Comment:
This would be wrong if the sqrt is an integer, no?

Reply:
Technically yes, but it doesn't matter if it's too big only too small

Reply:
It should at least be documented lest future readers stumble over this

Reply:
True, does this look better?